### DISCUSSION TITLE
rename  localhost to FW

### DISCUSSION THREAD

**Discussion by volga629-1** [FoobarOy/foomuuri#29] [Ideas]
The idea to have builtin zones like FW instead localhost.

**Comment by volga629-1** [comment]
Might be also VPN zone for different tunnels types.

**Comment by kimheino** [comment]
You can rename `localhost` zone to `fw` if you want to:

```
foomuuri {
  # call it "fw" instead of "localhost"
  localhost_zone  fw
}
```

Originally it was called "fw" in Foomuuri too. We decided to rename it to "localhost" after long discussions.

Foomuuri tries to be flexible on zone naming. That's why there are no builtin zones defined, except "localhost". You can easily define "vpn" and other zones:

```
# Define zones
zone {
  # If renamed, define "fw" instead of "localhost" here
  fw
  # add more zones
  wan
  lan
  vpn
  ...
}
```

For more information see wiki  [best practices](https://github.com/FoobarOy/foomuuri/wiki/Best-Practices#zone-names).

**Comment by volga629-1** [comment]
The idea is to have clear directions of traffic localhost mean 127.0.0.1 or loopback, but not firewall it self like FW.

**Comment by bleve** [comment]
I guess you have misunderstood what localhost means. All local addresses on firewall machine are local to host. So localhost in foomuuri means exactly same as fw in shorewall and has nothing to do dev lo or dev lo addresses.

Check: `ip -6 route list table local` and `ip route list table local` to get clarification.
